src/components/main/UsernameForm.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+
+import { ModeEdit } from '@mui/icons-material';
+import { Box, Grid, IconButton } from '@mui/material';
+
+import {
+  USERNAME_DISPLAY_ID,
+  USERNAME_EDIT_BUTTON_ID,
+} from '@/config/selectors';
+
+import { mutations } from '../../config/queryClient';
+import EditingUserNameField from './EditingNameField';
+
+type UsernameProps = {
+  member: {
+    id: string;
+    name: string;
+  };
+};
+const UsernameForm = ({ member }: UsernameProps): JSX.Element => {
+  const { mutate: editMember } = mutations.useEditMember();
+  const [isEditing, setIsEditing] = useState(false);
+
+  const handleSave = (newUserName: string) => {
+    editMember({
+      id: member.id,
+      name: newUserName,
+    });
+
+    setIsEditing(false);
+  };
+
+  const cancelEditing = () => {
+    setIsEditing(false);
+  };
+
+  if (isEditing) {
+    return (
+      <EditingUserNameField
+        name={member.name}
+        onSave={handleSave}
+        onCancel={cancelEditing}
+      />
+    );
+  }
+  return (
+    <Grid container alignItems="center" spacing={1}>
+      <Grid item>
+        <Box id={USERNAME_DISPLAY_ID}>{member.name}</Box>
+      </Grid>
+      <Grid item>
+        <IconButton
+          onClick={() => setIsEditing(true)}
+          id={USERNAME_EDIT_BUTTON_ID}
+        >
+          <ModeEdit />
+        </IconButton>
+      </Grid>
+    </Grid>
```

Comment:
In this case `Grid` is not the right choice I think. Prefer [`Stack`](https://mui.com/material-ui/react-stack/) as it allows for 1-dimensional arrangement of data, where grid is 2-dimensional. Also grid has some weird margin/padding behavior. 

```suggestion
    <Stack direction="row" alignItems="center" spacing={1}>
      <Box id={USERNAME_DISPLAY_ID}>{member.name}</Box>
      <IconButton
        onClick={() => setIsEditing(true)}
        id={USERNAME_EDIT_BUTTON_ID}
      >
        <ModeEdit />
      </IconButton>
    </Stack>
```
Also with stack you do not need to wrap the elements again in a `<Grid item ...`